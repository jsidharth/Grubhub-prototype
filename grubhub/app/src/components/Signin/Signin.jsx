import React, { Component } from 'react';
import { connect } from 'react-redux';
// import logo from './../../assets/grubhub-1177056.png';
import './style.css';
import { userActions } from '../../js/actions/index';
import { ToastContainer } from 'react-toastify';

class Signin extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
    };
  }
  handleChange = e => {
    this.setState({
      [e.target.id]: e.target.value,
    });
  };
  handleSignIn = e => {
    e.preventDefault();
    const payload = this.state;
    this.props.loginUser(payload);
  };
  render() {
    return (
      <div className="container shadow p-3 singup_form">
        {/* <img src={logo} className="rounded mx-auto d-block signinlogo" alt="Logo"></img> */}
        <form onSubmit={e => this.handleSignIn(e)}>
          <div className="form-group">
            <label htmlFor="email">Email address</label>
            <input
              type="email"
              className="form-control"
              id="email"
              aria-describedby="emailHelp"
              placeholder="Enter email"
              onChange={this.handleChange}
            ></input>
            <small id="emailHelp" className="form-text text-muted">
              We'll never share your email with anyone else.
            </small>
          </div>
          <div className="form-group">
            <label htmlFor="password">Password</label>
            <input
              type="password"
              className="form-control"
              id="password"
              placeholder="Password"
              onChange={this.handleChange}
            />
          </div>
          <button type="submit" className="btn btn-primary m-3">
            Sign in
          </button>
        </form>
        <ToastContainer autoClose={2000} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  loginUser: payload => dispatch(userActions.loginUser(payload, ownProps)),
});
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Signin);
