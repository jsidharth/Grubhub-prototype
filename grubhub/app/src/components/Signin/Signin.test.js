// Link.react.test.js
import React from 'react';
import {Signin} from './Signin';

import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Signin component', function(){
    it('should render correctly', () => {
      const component = shallow( <Signin/>);
      expect(component).toMatchSnapshot();
    });
    it('should be wrapped in a div', () => {
        const component = shallow( <Signin/>);
        expect(component.find('.singup_form').length).toEqual(1);
      });
    
      it('should have one Sign in button', () => {
        const component = shallow( <Signin/>);
        expect(component.find('.sign-in').length).toEqual(1);
      });
    
      it('should have email field', () => {
        const component = shallow( <Signin/>);
        expect(component.find('.email').length).toEqual(1);
      });
      it('should have password field', () => {
        const component = shallow( <Signin/>);
        expect(component.find('.password').length).toEqual(1);
      });

})

