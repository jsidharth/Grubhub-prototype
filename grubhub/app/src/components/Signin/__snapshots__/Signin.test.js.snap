// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin component should render correctly 1`] = `
<div
  className="container shadow p-3 singup_form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address
      </label>
      <input
        aria-describedby="emailHelp"
        className="form-control email"
        id="email"
        onChange={[Function]}
        placeholder="Enter email"
        required={true}
        type="email"
      />
      <small
        className="form-text text-muted"
        id="emailHelp"
      >
        We'll never share your email with anyone else.
      </small>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control password"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="btn btn-primary m-3 sign-in"
      type="submit"
    >
      Sign in
    </button>
  </form>
  <ToastContainer
    autoClose={2000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    role="alert"
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</div>
`;
